core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -437,7 +437,7 @@ object ProducerStateManager {
 
     val fileChannel = FileChannel.open(file.toPath, StandardOpenOption.CREATE, StandardOpenOption.WRITE)
     try fileChannel.write(buffer)
-    finally fileChannel.close()
+    finally fileChannel.force(true)
```

Comment:
I think we need to do force and then close.

Reply:
Raising fix.